service/qos/service_level_controller.cc

```diff
@@ -305,12 +305,18 @@ void service_level_controller::stop_legacy_update_from_distributed_data() {
     _global_controller_db->dist_data_update_aborter.request_abort();
 }
 
-future<std::optional<service_level_options>> service_level_controller::find_effective_service_level(auth::role_set roles, include_effective_names include_names) {
+future<std::optional<service_level_options>> service_level_controller::find_effective_service_level(const sstring& role_name) {
+    if (_sl_data_accessor->is_topology_state_upgraded()) {
+        co_return _effective_service_levels_db.contains(role_name) 
+            ? std::optional<service_level_options>(_effective_service_levels_db.at(role_name)) 
+            : std::nullopt;
```

Comment:
Nit: avoid double lookup by using iterators instead of the `contains` + `at` combo